schema/elements.go

```diff
@@ -255,6 +261,26 @@ func (s *Schema) Resolve(tr TypeRef) (Atom, bool) {
 		if !ok {
 			return Atom{}, false
 		}
+
+		// Allow field-level electives to override the referred type
+		switch {
```

Comment:
Should we fail if the sub-type is not a map?

Reply:
Yes and probably also should add that to schema validation